tests/snuba/search/test_backend.py

```diff
@@ -1217,6 +1229,221 @@ def test_first_release_environments(self):
+        #         env=[production]: nothing
+        #
+        #     first_release: 2
+        #         env=[]: A, C
+        #         env=[production, staging]: A
+        #         env=[staging]: nothing
+        #         env=[production]: A
+
+        # create some releases
+
+        release_1 = self.create_release(self.project, version="release_1")
+        release_2 = self.create_release(self.project, version="release_2")
+
+        # create an issue/group whose events that occur in 2 distinct environments
+
+        group_a_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_a'],
+                'event_id': 'aaa' + ('1' * 29),
+                'environment': 'example_staging'
+            },
+            project_id=self.project.id,
+        )
+
+        group_a_event_2 = self.store_event(
+            data={
+                'fingerprint': ['group_a'],
+                'event_id': 'aaa' + ('2' * 29),
+                'environment': 'example_production'
+            },
+            project_id=self.project.id,
+        )
+
+        group_a = group_a_event_1.group
+
+        # get the environments for group_a
+
+        prod_env = group_a_event_2.get_environment()
+        staging_env = group_a_event_1.get_environment()
+
+        # create an issue/group whose event that occur in no environments
+        # but will be tied to release_1
+
+        group_b_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_b'],
+                'event_id': 'bbb' + ('1' * 29)
+            },
+            project_id=self.project.id,
+        )
+        assert group_b_event_1.get_environment().name == u''  # has no environment
+
+        group_b = group_b_event_1.group
+
+        # create an issue/group whose event that occur in no environments
+        # but will be tied to release_2
+
+        group_c_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_c'],
+                'event_id': 'ccc' + ('1' * 29)
+            },
+            project_id=self.project.id,
+        )
+        assert group_c_event_1.get_environment().name == u''  # has no environment
+
+        group_c = group_c_event_1.group
+
+        # expect no groups since given release version does not exist
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % 'fake',
+        )
+
+        assert set(results) == set([])
+
+        # expect no groups even though there are releases; since no group
+        # is attached to any releases
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([])
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_2.version,
+        )
+        assert set(results) == set([])
+
+        # set group_a's first_release to be release_1
+
+        group_a.first_release = release_1
```

Comment:
You could pass the `release` attribute into `store_event` to save assigning the release down here and re-saving the group.